### PR TITLE
Add support of out of context designs

### PR DESCRIPTION
**Pull request by ttown523**
This pull request adds support for out of context design checkpoints. It also updates the function `CellNet::computeRouteStatus()` to fix a few bugs and more accurately represents the routing state of an imported net.

### CODE REVIEW

File: src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellNet.java [hunk 1]
```diff
@@ -798,6 +813,7 @@ public RouteTree getSinkRouteTree(SitePin sitePin) {
 			return Collections.emptyList();
 		}
 		
+		// TODO: use an EntrySet instead of a KeySet
```
Comment (trharoldsen) [inline src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellNet.java]: Explain this TODO

File: src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellNet.java [hunk 2]
```diff
@@ -724,6 +732,13 @@ public RouteTree getIntersiteRouteTree() {
 	}
 	
 	/**
+	 * Returns the number of intersite route trees connected to this net
```
Comment (trharoldsen) [inline src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellNet.java]: missing period

**Comment by ttown523** [comment]
I have addressed your changes and will include them in the pull request I am submitting later today because I am lazy.

**Comment by ttown523** [comment]
@trharoldsen can you approve this if there is no other requested changes?

**Pull request merged**